### PR TITLE
API-189: Allow the nunit runner to set the number of test agents

### PR DESCRIPTION
**Pull request by kieranlynam**
http://youtrack.zopian.ad.zopa.com/issue/API-189

This will allow us to run test projects in sequence in certain repos. See https://github.com/nunit/docs/wiki/Parallel-Test-Execution

:mag: @xiaoqingzhang @lukesmith @lavapatch @vytautasziurlis

### CODE REVIEW

File: lib/bozo/test_runners/nunit3.rb [hunk 1]
```diff
@@ -57,6 +61,10 @@ def runner_args
         args << '--x86'
       end
 
+      unless @agents.blank?
+        args << '--agents=#{@agents}'
```
Comment (stephenbinns) [inline lib/bozo/test_runners/nunit3.rb]: You've not used the correct string here you need to use double quotes to get string interpolation

Comment (kieranlynam) [inline lib/bozo/test_runners/nunit3.rb]: Is this is a ghost????

Comment (stephenbinns) [inline lib/bozo/test_runners/nunit3.rb]: No just a bug :p

File: lib/bozo/test_runners/nunit3.rb [hunk 2]
```diff
@@ -57,6 +61,10 @@ def runner_args
         args << '--x86'
       end
 
+      unless @agents.blank?
```
Comment (stephenbinns) [inline lib/bozo/test_runners/nunit3.rb]: Also this method doesn't exist in the standard ruby library its a rails method  you want something like 

`agents && agents.empty?`

**Comment by xiaoqingzhang** [comment]
👍 LGTM

**Comment by xiaoqingzhang** [comment]
👍 LGTM

File: lib/bozo/test_runners/nunit3.rb [hunk 3]
```diff
@@ -57,6 +61,10 @@ def runner_args
         args << '--x86'
       end
 
+      unless !agents || agents.empty?
```
Comment (lukesmith) [inline lib/bozo/test_runners/nunit3.rb]: this should be `unless !@agents || @agents.empty?` or you'll call the `agents` method with no arguments

Comment (kieranlynam) [inline lib/bozo/test_runners/nunit3.rb]: https://github.com/zopaUK/bozo-scripts/pull/73

**Pull request merged**